prog/weavedns/main.go

```diff
@@ -95,14 +96,18 @@ func main() {
 	if err != nil {
 		Error.Fatal("[main] Unable to initialize the Zone database", err)
 	}
-	err = zone.Start()
-	if err != nil {
+	if err := zone.Start(); err != nil {
 		Error.Fatal("[main] Unable to start the Zone database", err)
 	}
 	defer zone.Stop()
 
+	dockerCli, err = docker.NewClient(apiPath)
+	if err != nil {
+		Error.Fatal("[main] Unable to start docker client", err)
+	}
```

Comment:
This is a change in semantics - the point of fetching the version from Docker was to check that we could actually communicate, otherwise the AddObserver() silently fails to do anything.